cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -396,3 +397,13 @@ func getSpotVMOptions(spotVMOptions *infrav1.SpotVMOptions) (compute.VirtualMach
 	}
 	return compute.Spot, compute.Deallocate, billingProfile, nil
 }
+
+func getSecurityProfile(vmSpec azure.VMSpec) *compute.SecurityProfile {
+	if vmSpec.SecurityProfile == nil {
+		return nil
+	}
+
+	return &compute.SecurityProfile{
+		EncryptionAtHost: to.BoolPtr(*vmSpec.SecurityProfile.EncryptionAtHost),
```

Comment:
this isn't supported on all VM sizes, it requires a check against the capability (see accelerated networking for an example). 

we should probably fail fast with error + no requeue in the reconciler if it's not possible to provision that for the user.

Reply:
I added the capability check.